code/game/objects/items/shuttle_creator.dm

```diff
@@ -0,0 +1,301 @@
+		return
+	to_chat(user, "<span class='warning'>The [src] bleeps. Select an airlock to create a docking port, or a valid machine to link.</span>")
+	return
+
+/obj/item/shuttle_creator/proc/calculate_bounds(obj/docking_port/mobile/port)
+	if(!port || !istype(port, /obj/docking_port/mobile))
+		return FALSE
+	//Heights is the distance away from the port
+	//width is the distance perpendicular to the port
+	var/minX = INFINITY
+	var/maxX = 0
+	var/minY = INFINITY
+	var/maxY = 0
+	for(var/turf/T in loggedTurfs)
+		minX = min(T.x, minX)
+		maxX = max(T.x, maxX)
+		minY = min(T.y, minY)
+		maxY = max(T.y, maxY)
+	//Make sure shuttle was actually found
+	if(maxX == INFINITY || maxY == INFINITY)
+		return FALSE
+	minX--
+	minY--
+	var/width = maxX - minX
+	var/height = maxY - minY
+	var/offset_x = port.x - minX
+	var/offset_y = port.y - minY
+	switch(port.dir) //Source: code/datums/shuttles.dm line 77 (14/03/2020) :)
+		if(NORTH)
+			port.width = width
+			port.height = height
+			port.dwidth = offset_x - 1
+			port.dheight = offset_y - 1
+		if(EAST)
+			port.width = height
+			port.height = width
+			port.dwidth = height - offset_y
+			port.dheight = offset_x - 1
+		if(SOUTH)
+			port.width = width
+			port.height = height
+			port.dwidth = width - offset_x
+			port.dheight = height - offset_y
+		if(WEST)
+			port.width = height
+			port.height = width
+			port.dwidth = offset_y - 1
+			port.dheight = width - offset_x
+	to_chat(usr, "Created shuttle with dir [port.dir], width [port.width], height [port.height], dwidth [port.dwidth], dheight [port.dheight]")
+	return TRUE
+
+//Go through all the all_turfs and check which direction doesn't have the shuttle
+/obj/item/shuttle_creator/proc/getNonShuttleDirection(turf/targetTurf)
+	var/position = null
+	if(!(get_offset_target_turf(targetTurf, 0, 1) in loggedTurfs))
+		if(position != null)
+			return null
+		position = NORTH
+	if(!(get_offset_target_turf(targetTurf, 0, -1) in loggedTurfs))
+		if(position != null)
+			return null
+		position = SOUTH
+	if(!(get_offset_target_turf(targetTurf, 1, 0) in loggedTurfs))
+		if(position != null)
+			return null
+		position = EAST
+	if(!(get_offset_target_turf(targetTurf, -1, 0) in loggedTurfs))
+		if(position != null)
+			return null
+		position = WEST
+	to_chat(usr, "Direction to space is [position]")
+	return position
+
+/obj/item/shuttle_creator/proc/invertDir(var/input_dir)
+	if(input_dir == NORTH)
+		return SOUTH
+	else if(input_dir == SOUTH)
+		return NORTH
+	else if(input_dir == EAST)
+		return WEST
+	else if(input_dir == WEST)
+		return EAST
+	return null
+
+/obj/item/shuttle_creator/proc/shuttle_create_docking_port(atom/target, mob/user)
+
+	if(loggedTurfs.len == 0 || !recorded_shuttle_area)
+		to_chat(user, "<span class='warning'>Invalid shuttle, restarting bluespace systems...</span>")
+		return FALSE
+
+	var/datum/map_template/shuttle/new_shuttle = new /datum/map_template/shuttle()
+	var/static/idnum = 0
```

Comment:
Why is this static?

Reply:
Fixed, was from the previous static variable